CONTRIBUTING.md

```diff
@@ -10,6 +10,16 @@ This repository is HASH's public monorepo. It contains [several different](READM
 
 To ascertain the license and contributor policy for a given project, check out the `LICENSE.md` and `CONTRIBUTOR.md` files in its root.
 
+## Common contribution processes
+
+These apply across all projects:
+
+- Before undertaking any significant work, please share your proposal with us. You can do so as a [discussion](https://github.com/hashintel/hash/discussions) if it's a feature proposal, or an [issue](https://github.com/hashintel/hash/issues) if it's a bug you intend to fix. If you're unsure, you can always chat to us on [Discord](https://hash.ai/discord) first. If it's a typo or docs change, don't worry about this step.
+- When submitting a pull request, please fill out any sections of the provided template you feel able to. If you are unsure or don't feel a section is relevant, please say so. Include a link to the issue or discussion proposing the change.
+- Once you have receive a pull request review, please bear the following in mind:
+  - reviewers may make suggestions for _optional_ changes which are not required to get your code merged. It should be obvious which suggestions are obvious, and which are required changes. If it is not obvious, ask for clarification.
+  - please only resolve comment threads when it is obvious there is no scope for further discussion. If in doubt, leave them open. If comment threads discuss work to be done in follow up, always leave them open (so they can be directly linked to).
```

Comment:
I think we might want to be even more prescriptive. In that, the OP of the comment thread should be the only one to resolve it, that is, you should not be resolving comment threads you didn't open, this allows reviewers to be confident in your changes and get things through faster. The exception to this is if there was a suggested change using the Github features for suggesting changes in the comment. In that case, resolving it directly is fine as there's no need for further discussion.

Reply:
> was a suggested change using the Github features for suggesting changes in the comment

This is gross but I don't know how to describe it in a clearer way

Reply:
Here's a version: https://github.com/hashintel/hash/pull/345/commits/62fa2c59bb1c45e3f8e4e9ede6ed0489db70ef97

Reply:
I think it's a lot better. Not sure it's clear that we're fine with suggestion threads being resolved, it sort of reads that you might need to undo Github's auto-resolve but I don't think we need to strive for perfection here